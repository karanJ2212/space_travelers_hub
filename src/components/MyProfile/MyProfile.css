.profile-section {
  display: flex;
  flex-direction: row;
  gap: 15rem;
  margin-top: 2rem;
  justify-content: center;
}

h3 {
  font-family: Arial, Helvetica, sans-serif;
}

.rocket-line {
  height: 2px;
  width: fit-content;
  background-color: black;
  margin-top: 3px;
}

/*
.table {
  margin-top: 3rem;
  padding-top: 2rem;
} */

.table-rocket {
  border: 2px solid black;
  border-collapse: collapse;
  width: fit-content;
}

.table-rocket th,
.table-rocket td {
  border: 1px solid black;
  padding: 8px;
  text-align: left;
}

.border-rocket {
  border: 1px solid black;
}

.rocket-paragraph {
  margin: 10px;
}
