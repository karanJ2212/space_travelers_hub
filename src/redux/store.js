import { configureStore } from '@reduxjs/toolkit';
import rocketReducer from './rockets/rocketsSlice';
import missionReducer from './missions/missionslice';

const store = configureStore({
  reducer: {
    rockets: rocketReducer,
    missions: missionReducer,
  },
});

export default store;
