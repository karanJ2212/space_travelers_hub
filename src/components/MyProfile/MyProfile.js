import React from 'react';
import { useSelector } from 'react-redux';
import './MyProfile.css';

export default function MyProfile() {
  const rockets = useSelector((state) => state.rockets.rockets);
  const missions = useSelector((state) => state.missions.missions);
  return (
    <div className="profile-section">
      <div className="mprdiv">
        <h2 className="my-profile-h2">My missions</h2>
        <table className="table">
          <tbody>
            {missions
              .filter((missions) => missions.reserved)
              .map((missions) => (
                <tr key={missions.mission_id} className="border">
                  <td className="profiletd">{missions.mission_name}</td>
                </tr>
              ))}
          </tbody>
        </table>
      </div>
      <div className="mprdiv">
        <h2 className="my-profile-h2">My rockets</h2>
        <table className="table">
          <tbody>
            {rockets
              .filter((rockets) => rockets.reserved === true)
              .map((rockets) => (
                <tr key={rockets.id} className="border">
                  <td className="profiletd">{rockets.rocketName}</td>
                </tr>
              ))}
          </tbody>
        </table>
      </div>
    </div>
  );
}
