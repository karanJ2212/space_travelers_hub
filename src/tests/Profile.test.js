import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import MyProfile from '../components/MyProfile/MyProfile';
import store from '../redux/store';

it('matches snapshot', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <MyProfile />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
