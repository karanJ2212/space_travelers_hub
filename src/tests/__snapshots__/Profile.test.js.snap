// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="profile-section"
>
  <div
    className="mprdiv"
  >
    <h2
      className="my-profile-h2"
    >
      My missions
    </h2>
    <table
      className="table"
    >
      <tbody />
    </table>
  </div>
  <div
    className="mprdiv"
  >
    <h2
      className="my-profile-h2"
    >
      My rockets
    </h2>
    <table
      className="table"
    >
      <tbody />
    </table>
  </div>
</div>
`;
